plutus-tx/src/PlutusTx/IsData/TH.hs

```diff
@@ -103,7 +105,7 @@ unsafeFromDataClause indexedCons = do
     let cases =
             foldl'
             (\kont ixCon -> unsafeReconstructCase ixCon (TH.varE indexName) [| BI.snd $(TH.varE tupName) |] kont)
-            [| Builtins.error () |]
+            [| traceError $ fromString "UD" |]
```

Comment:
What does UD mean? I guess we don't have a registry of such strings, but we should at least have a comment here.

Reply:
"Unsafe Data", from "unsafeFromDataClause"

I don't know to clarify it. Impossible case? And there is a comment above about the error.

Other codes does make little sense (to me at least) unfortunately as well. Maybe create a module with all codes as variables?

Reply:
Great idea. `PlutusTx.ErrorCodes`? This can be the first one! Then we could look through `plutus-tx` and find all the other uses of `traceError` and replace them with codes. I know there are some in `plutus-ledger` too, but they'll have to document them separately.

Reply:
And then I'd suggest something like `PT01` or something (PT=plutus-tx, and a numeric suffix)

Reply:
I just discovered https://github.com/input-output-hk/plutus/blob/master/plutus-core/common/ErrorCode.hs

Shouldn't we use that?

Reply:
That's for errors emitted by the tools, not by the functions themselves in on-chain code. I think we can have something else (and anyway that one is not quite completely implemented...).